Sources/XiEditor/RPCSending.swift

```diff
@@ -222,78 +257,85 @@ class StdoutRPCSender: RPCSending {
     }
 
     private func handleRequest(json: [String: AnyObject]) {
-        guard let method = json["method"] as? String, let params = json["params"], let id = json["id"]
+        guard let jsonMethod = json["method"] as? String,
+              let params = json["params"],
+              let id = json["id"],
+              let method = RPCRequestMethod(rawValue: jsonMethod)
             else {
-                print("unknown json from core: \(json)")
+                assertionFailure("unknown json from core: \(json)")
                 return
         }
 
         switch method {
-        case "measure_width":
-            let args = params as! [[String: AnyObject]]
-            if let result = self.client?.measureWidth(args: args) {
-                self.sendResult(id: id, result: result)
+        case .measureWidth:
+            guard let args = params as? [[String: AnyObject]],
+                  let result = client?.measureWidth(args: args) else {
+                    assertionFailure("unexpected data from core: \(params)")
+                    return
             }
-        default:
-            print("unknown request \(method)")
+            
```

Comment:
hard to tell from the diff but this looks like unintended whitespace

Reply:
![](https://user-images.githubusercontent.com/7672056/50523423-38d09f00-0ab7-11e9-9db6-450d06256180.png)
I think I've added this line, after the `guard` statement.

Reply:
I was looking at the whitespace on the line above that? Obviously no big deal but I like turning on 'trim trailing whitespace' + 'including blank lines' in Xcode.